File: lib/models/search_result.dart
class SearchResult {
  final String videoId;
  final String title;
  final String author;
  final String thumbnailUrl;
  final Duration duration;

  SearchResult({
    required this.videoId,
    required this.title,
    required this.author,
    required this.thumbnailUrl,
    required this.duration,
  });

  int get durationInSeconds => duration.inSeconds;
}

File: react-music-app/src/models/SearchResult.ts
import { Song } from './Song';

export interface SearchResult {
  videoId: string;
  title: string;
  author: string;
  thumbnailUrl: string;
  durationSeconds: number;
}

export function searchResultToSong(result: SearchResult): Song {
  return {
    id: result.videoId,
    title: result.title,
    artist: result.author,
    thumbnailUrl: result.thumbnailUrl,
    duration: result.durationSeconds,
    isDownloaded: false,
    createdAt: new Date().toISOString(),
  };
}
